tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/TestClassInfoRunClassInitializeIntegration.cs

```diff
@@ -0,0 +1,83 @@
+// <copyright file="TestClassInfoRunClassInitializeIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.ComponentModel;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using Datadog.Trace.Ci;
+using Datadog.Trace.ClrProfiler.CallTarget;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.MsTestV2;
+
+/// <summary>
+/// Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.Execution.TestClassInfo.RunClassInitialize(TestContext testContext) calltarget instrumentation
+/// </summary>
+[InstrumentMethod(
+    AssemblyName = "Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter",
+    TypeName = "Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.Execution.TestClassInfo",
+    MethodName = "RunClassInitialize",
+    ReturnTypeName = ClrNames.Void,
+    ParameterTypeNames = new[] { "_" },
+    MinimumVersion = "14.0.0",
+    MaximumVersion = "14.*.*",
+    IntegrationName = MsTestIntegration.IntegrationName)]
+[Browsable(false)]
+[EditorBrowsable(EditorBrowsableState.Never)]
+public static class TestClassInfoRunClassInitializeIntegration
+{
+    private static readonly MethodInfo EmptyCleanUpMethodInfo = typeof(TestAssemblyInfoRunAssemblyInitializeIntegration).GetMethod("EmptyCleanUpMethod", BindingFlags.NonPublic | BindingFlags.Static);
+    internal static readonly ConditionalWeakTable<object, object> TestClassInfos = new();
+
+    /// <summary>
+    /// OnMethodBegin callback
+    /// </summary>
+    /// <typeparam name="TTarget">Type of the target</typeparam>
+    /// <typeparam name="TContext">Type of the ITestMethod</typeparam>
+    /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+    /// <param name="testContext">Test context instance</param>
+    /// <returns>Calltarget state value</returns>
+    internal static CallTargetState OnMethodBegin<TTarget, TContext>(TTarget instance, TContext testContext)
+        where TTarget : ITestClassInfo
+    {
+        if (!MsTestIntegration.IsEnabled)
+        {
+            return CallTargetState.GetDefault();
+        }
+
+        if (!TestClassInfos.TryGetValue(instance.Instance, out var suiteObject))
+        {
+            instance.TestCleanupMethod ??= EmptyCleanUpMethodInfo;
```

Comment:
Is this necessary? If it's null, you're setting it to a method that doesn't seem to do anything

Reply:
Got it, it didn't occur to me that the integration to "close" the module/suite was dependent on the cleanup event. Thanks for clarifying